facilitator/src/bin/facilitator.rs

```diff
@@ -528,18 +537,11 @@ fn main() -> Result<(), Error> {
             let share_processor_key_bytes =
                 base64::decode(sub_matches.value_of("share-processor-private-key").unwrap())
                     .unwrap();
-            let share_processor_key = match EcdsaKeyPair::from_pkcs8(
+            let share_processor_key = EcdsaKeyPair::from_pkcs8(
                 &ECDSA_P256_SHA256_FIXED_SIGNING,
                 &share_processor_key_bytes,
-            ) {
-                Ok(priv_key) => priv_key,
-                Err(e) => {
-                    return Err(Error::CryptographyError(
-                        "failed to parse value for share-processor-private-key".to_owned(),
-                        e,
-                    ))
-                }
-            };
+            )
+            .context("failed to parse value for share-processor-private-key")?;
```

Comment:
I love all the places where you replace seven lines of matching with a one line call.

Reply:
There's a lot more places I could've done that, but they'd require some restructuring of the code usually, so I decided to just not do any of that at all in order to avoid going into rabbit holes. So, future work!